apps/drain-safe/src/App.tsx

```diff
@@ -0,0 +1,209 @@
+  const services = useServices(safe.network);
+  const web3: Web3 | undefined = services?.web3;
+  const [submitting, setSubmitting] = useState(false);
+  const [assets, setAssets] = useState<Asset[]>([]);
+  const [toAddress, setToAddress] = useState<string>('');
+  const [finishedTxHash, setFinishedTxHash] = useState<string>('');
+
+  const fetchBalances = useCallback(async (): Promise<void> => {
+    // Fetch safe assets
+    try {
+      const data = await fetchSafeAssets(safe.safeAddress, safe.network);
+      setAssets(data.items);
+    } catch (err) {
+      console.error("Couldn't load assets", err);
+    }
+  }, [safe]);
+
+  const encodeTxData = useCallback(
+    (recipient, amount): string => {
+      if (web3 == null) {
+        return '0x';
+      }
+
+      return web3.eth.abi.encodeFunctionCall(erc20.transfer as AbiItem, [
+        web3.utils.toChecksumAddress(recipient),
+        amount,
+      ]);
+    },
+    [web3],
+  );
+
+  const submitTx = useCallback(async () => {
+    setSubmitting(true);
+
+    if (!web3!.utils.isAddress(toAddress)) {
+      return;
+    }
+
+    const txs = assets.map((item) => {
+      return item.tokenInfo.type === 'ETHER'
+        ? {
+            // Send ETH directly to the recipient address
+            to: web3!.utils.toChecksumAddress(toAddress),
+            value: item.balance,
+            data: '0x',
+          }
+        : {
+            // For other token types, generate a contract tx
+            to: web3!.utils.toChecksumAddress(item.tokenInfo.address),
+            value: '0',
+            data: encodeTxData(toAddress, item.balance),
+          };
+    });
+
+    let safeTxHash = '';
+    try {
+      const data = await sdk.txs.send({ txs });
+      safeTxHash = data.safeTxHash;
+      console.log(safeTxHash);
+    } catch (e) {
+      console.error(e);
+    }
+
+    if (!safeTxHash) {
+      setSubmitting(false);
+      return;
+    }
+
+    const poll = setInterval(async (): Promise<void> => {
+      try {
+        const safeTx = await sdk.txs.getBySafeTxHash(safeTxHash);
+        console.log(safeTx);
+      } catch (e) {
+        return;
+      }
+      setSubmitting(false);
+      setFinishedTxHash(safeTxHash);
+
+      setAssets(
+        assets.map((item) => ({
+          ...item,
+          balance: '0',
+          fiatBalance: '0',
+        })),
+      );
+    }, 1000);
+
+    return () => {
+      clearInterval(poll);
+    };
+  }, [sdk, assets, toAddress, web3, encodeTxData]);
+
+  const onToAddressChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+    setToAddress(e.target.value);
+  };
+
+  // Fetch balances
+  useEffect(() => {
+    fetchBalances();
+  }, [fetchBalances]);
+
+  return (
+    <Container>
+      <Flex>
+        <Title size="md">Drain Account</Title>
+        <div style={{ flex: 1 }} />
+        <EthHashInfo hash={safe.safeAddress} network={safe.network} textSize="lg" showIdenticon showCopyBtn />
+      </Flex>
+
+      {finishedTxHash && (
```

Comment:
The transaction hash returned is a safe transaction hash, not the actual ethereum transaction hash, thus etherscan won't work. More info: https://docs.gnosis.io/safe/docs/contracts_tx_execution/#transaction-hash

Think of it as of internal transaction hash used by safe contracts to distinguish transactions. Also, when you approve a transaction in the interface, you actually approve the hash.

Reply:
Oh, OK. I guess it doesn't make sense to display it then?

Reply:
Yes, etherscan link will be broken and confusing for the user. Only nerds will get it 😀